fw/hpack.c

```diff
@@ -3013,10 +3072,11 @@ tfw_hpack_rbuf_calc(TfwHPackETbl *__restrict tbl, unsigned short new_size,
 	do {
 		unsigned short f_len, fhdr_len;
 
-		if (i >= HPACK_MAX_ENC_EVICTION)
+		if (i >= HPACK_MAX_ENC_EVICTION && del_list)
```

Comment:
I believe we must get rid of `HPACK_MAX_ENC_EVICTION` limitation as HTTP/2 spec explicitly mentions that there is "typical" use case is when peer send 2 sequential hdr table size updates: first with 0 value, followed by the new non-zero value just to completely purge headers table.

We should either add some hack or make an intrusive list of "headers to be evicted" avoiding any temporary arrays.

Reply:
I don't see problem in HPACK_MAX_ENC_EVICTION. When peer send sequential hdr table size updates: first with 0 value, followed by the new non-zero value just to completely purge headers table we pass del_list == 0 and table will be cleared. del_list is not zero only when we add new entry in the table!